dask_mongo/core.py

```diff
@@ -14,19 +18,68 @@
 
 appname = f"dask-mongo/{__version__}"
 
+_CACHE_SIZE = 10
+_CLIENTS = weakref.WeakValueDictionary({})
+
+
+def _recursive_tupling(item):
+    if isinstance(item, list):
+        return tuple([_recursive_tupling(i) for i in item])
+    if isinstance(item, Mapping):
+        return tuple(
+            [(_recursive_tupling(k), _recursive_tupling(v)) for k, v in item.items()]
+        )
+    elif isinstance(item, Hashable):
+        return hash(item)
+    else:
+        return item
+
+
+class HashableKwargs(dict):
```

Comment:
I'd suggest removing this new type altogether. Instead of calling hash() just use a single function that recursively makes kwargs hashable, perhaps "_hashable_kwargs()" or "_freeze(**kwargs)".